tsdb/engine/tsm1/engine.go

```diff
@@ -1227,9 +1229,59 @@ func (e *Engine) compactTSMFull(quit <-chan struct{}) {
 				// Release the files in the compaction plan
 				e.CompactionPlan.Release(s.compactionGroups)
 			}
+		}
+	}
+}
+
+// onFileStoreReplace is callback handler invoked when the FileStore
+// has replace one set of TSM files with a new set.
+func (e *Engine) onFileStoreReplace(newFiles []TSMFile) {
+	// Load any new series keys to the index
+	readers := make([]chan seriesKey, 0, len(newFiles))
+	for _, r := range newFiles {
+		ch := make(chan seriesKey, 1)
+		readers = append(readers, ch)
+
+		go func(c chan seriesKey, r TSMFile) {
+			n := r.KeyCount()
```

Comment:
`KeyCount` ends up taking `RLocks` on the reader and I didn't want to lock the readers with every loop iteration.